### PR TITLE
Split repo into components

### PR DESCRIPTION
**Pull request by jayzes**
### Are there any dependencies (software or human) that need to be addressed before this PR is merged?
Nope

### What ticket(s) or other PRs does this relate to?
None

### What was the problem or feature?
This gem by default requires a number of components (ActiveRecord, Datadog, Sentry) that may not be universally applicable.

### What was the solution?
Split into multiple gems so consumers can pick and choose what they want to use.

- [x] Security Impact has been considered: None
- [x] Network Impacts have been considered: None

### Where does this work fall on the Good - Fast spectrum?
🐢 

### Any additional work needed as a result of merging this PR (deploy steps, other PRs, etc.)?
Setting up new gems for release

### CODE REVIEW

**Comment by jayzes** [comment]
@ianbender @plukevdh would love your thoughts on this before I go much further. The other option would be to do multiple repos, but that seemed like a lot.

**Comment by plukevdh** [comment]
I can dig this. I'd love to see it setup the way AWS does their single repo/multi library repo so that we can build/publish the individual components as independently requirable gems: https://bundler.io/guides/git.html

**Comment by jayzes** [comment]
@plukevdh for sure! I was kind of thinking the way that Rails has its repo structured

**Comment by plukevdh** [comment]
> I was kind of thinking the way that Rails has its repo structured

What is... Rails?

**Comment by jayzes** [comment]
I'll still need to add workflow to release the sub-gems, but the rest is ready for a final look. One outstanding question is that this is a breaking change, and I'm not 100% sure how to handle the transition. I could write a legacy shim to accommodate existing uses or keep the old stack in place and create a new opt-in one for modular use. Any thoughts on the right approach there? We are currently using the versioned gem (at least in `lambdas`), so we could try to manage the transition using semver.

File: README.md [hunk 1]
```diff
@@ -54,10 +54,46 @@ Existing middleware include:
...
+- `SentryMonitoring` - instrument your function code for Sentry's performance monitoring (`require "airfoil/middleware/sentry_monitoring"`)
+```ruby
+b.use Airfoil::Middleware::SentryCatcher
+```
+
+- `SentryMonitoring` - instrument your function code for Sentry's performance monitoring (`require "airfoil/middleware/sentry_monitoring"`)
+```ruby
+b.use Airfoil::Middleware::SentryCatcher
+```
+
```
Comment (ianbender) [inline README.md]: Looks like this bit is duplicated. Also, the `SentryCatcher` middleware is show in the code samples both on the `SentryMonitoring` and `Datadog` bits.

File: airfoil/lib/airfoil.rb [hunk 2]
```diff
@@ -26,15 +21,17 @@ def create_stack
         puts "Received SIGTERM, shutting down gracefully..." # rubocop:disable Rails/Output
       end
 
+      logger = defined?(::Rails) ? Rails.logger : Logger.new($stdout, level: (ENV["LOG_LEVEL"] || :info).to_sym)
+
       ::Middleware::Builder.new { |b|
-        b.use Middleware::LoggerTagging, Rails.logger
+        if defined?(::Rails)
+          b.use Middleware::LoggerTagging, logger
```
Comment (ianbender) [inline airfoil/lib/airfoil.rb:28]: With logger defined above, do we still want on use it here only if running within Rails?

**Comment by ianbender** [comment]
@jayzes I'm not sure where we're using this outside of the `lambdas` repo. If that's the only spot then I think it would be okay to try and manage the rollout based on semver but @plukevdh would likely have more context on other repos.

File: airfoil/lib/airfoil.rb (hunk 2 above)
Comment (jayzes) [reply airfoil/lib/airfoil.rb:28]: The tagged logging functionality is in ActiveSupport (https://api.rubyonrails.org/classes/ActiveSupport/TaggedLogging.html#method-i-tagged) - we could make the middleware optional I suppose, and that way apps not using Rails but using Ruby with some other configuration for a global logger could still use this bit?

File: README.md (hunk 1 above)
Comment (jayzes) [reply README.md]: Oops, good catches. Updated!

**Comment by plukevdh** [comment]
AFAIK right now, `lambdas` is all that is using this. I'd love to see it utilized wider. Semver should be good so long as it's tag-based rather than just a `VERSION` change. Usage of a version tag needs a gem repo release flow to make use of. Probably either GH packages or Rubygems/private repo of some kind

File: airfoil/lib/airfoil.rb (hunk 2 above)
Comment (plukevdh) [reply airfoil/lib/airfoil.rb:28]: FYI, the Datadog logger requires some version of a tagged logger. Not sure if this will impact or not.

**Review by plukevdh — APPROVED** [review]
Looks great. Seems like there might be a way to clean up some of the duplication in the gemspecs and associated files but i know other repos are doing the same thing. Some generators might be appropriate. Otherwise 👍🏻

**Comment by jayzes** [comment]
I'll go ahead and merge this one so we can build on top of it

**Pull request merged**